model/simulationtests/schedule_file_2.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+# This test aims to test the new ScheduleFile ctor that takes a direct path and does not make a copy to the companion directory. Cf NREL/OpenStudio#4857
+
+require 'openstudio'
+require_relative 'lib/baseline_model'
+
+model = BaselineModel.new
+
+# make a 1 story, 100m X 50m, 1 zone building
+model.add_geometry({ 'length' => 100,
+                     'width' => 50,
+                     'num_floors' => 1,
+                     'floor_to_floor_height' => 4,
+                     'plenum_height' => 0,
+                     'perimeter_zone_depth' => 0 })
+
+# add windows at a 40% window-to-wall ratio
+model.add_windows({ 'wwr' => 0.4,
+                    'offset' => 1,
+                    'application_type' => 'Above Floor' })
+
+# add thermostats
+model.add_thermostats({ 'heating_setpoint' => 19,
+                        'cooling_setpoint' => 26 })
+
+# assign constructions from a local library to the walls/windows/etc. in the model
+model.set_constructions
+
+# set whole building space type; simplified 90.1-2004 Large Office Whole Building
+model.set_space_type
+
+# add design days to the model (Chicago)
+model.add_design_days
+
+# In order to produce more consistent results between different runs,
+# we sort the zones by names
+# (There's only one here, but just in case this would be copy pasted somewhere
+# else...)
+zones = model.getThermalZones.sort_by { |z| z.name.to_s }
+z = zones[0]
+z.setUseIdealAirLoads(true)
+
+###############################################################################
+
+# Add output variables?
+add_out_vars = false
+
+###############################################################################
+
+# set year to 2013
+yd = model.getYearDescription
+yd.setCalendarYear(2013)
+
+# add schedule file
+file_name = File.join(File.dirname(__FILE__), 'lib/schedulefile.csv')
+file_name = File.realpath(file_name)
+schedule_file = OpenStudio::Model::ScheduleFile.new(model, file_name, 3, 1)
```

Comment:
Why doesn't this line fail? The new ScheduleFIle ctor takes an openstudio path, not a string...

Reply:
(C++) Implicit conversion from std::string to path. See MCVE https://gcc.godbolt.org/z/79qYxd5e4